### PR TITLE
Add molecular structure support

### PR DESCRIPTION
**Pull request by kroenlein**
Adding support to GEMD for SMILES- and InChI-formatted molecular structures.

As per the designs communicated in 

https://citrineinformatics.github.io/gemd-docs/specification/value-types/

this implements and tests those structures.

https://citrine.atlassian.net/browse/PLA-1876

### CODE REVIEW

File: gemd/entity/value/smiles_value.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"""An empirical chemical formaula."""
+from gemd.entity.value.molecular_value import MolecularValue
+
+
+class SMILES(MolecularValue):
```
Comment (kroenlein) [inline gemd/entity/value/smiles_value.py:5]: @djack What is Citrine Best Practice for acronyms in class names?  Should this be `SMILES`, `Smiles` or other?

Comment (sparadiso) [reply gemd/entity/value/smiles_value.py:5]: It's `Smiles` in the backend, fwiw.

File: setup.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def run(self):
 
 
 setup(name='gemd',
-      version='0.6.6',
+      version='0.6.7',
```
Comment (bfolie) [inline setup.py:39]: Minor version bump because it's a new feature?

Comment (kroenlein) [reply setup.py:39]: The deprecation warnings on the Taurus migration are:
```
./gemd/client/json_encoder.py:@deprecation.deprecated(deprecated_in="0.6.0", removed_in="0.7.0",
```
So, from a process perspective, I should be wiping out the Taurus tree in a different PR if we're doing the minor version bump.

Comment (bfolie) [reply setup.py:39]: ack

Well I don't know what the best process is... but this PR itself looks fine.

**Pull request merged**